api/controllers/OpenEAppController.js

```diff
@@ -96,7 +100,31 @@ const OpenEAppController = {
     },
 
     _formatDate(date) {
-        return moment(date).format('DD MMMM YYYY');
+        return dayjs(date).format('DD MMMM YYYY');
+    },
+
+    _calculateDaysLeftToDownload(req) {
+        const urlObj = new URL(
+            `${req.protocol}://${req.headers.host}${req.originalUrl}`
+        );
+        const queryParams = new URLSearchParams(urlObj.search);
+        if (!queryParams.has('completedDate')) {
+            throw new Error('No date value found in url');
+        }
+        const completedDate = queryParams.get('completedDate');
+        const currentDate = dayjs(Date.now());
+        const differenceBetweenCurrentAndCompletedDate = currentDate.diff(
+            completedDate,
+            'day'
+        );
+        const daysLeftToDownloadDoc =
+            MAX_DAYS_TO_LEGALISE -
+            Number(differenceBetweenCurrentAndCompletedDate);
+        const numberIsNegative = Math.sign(daysLeftToDownloadDoc) === -1;
+        if (numberIsNegative) {
+            throw new Error('Application has expired');
+        }
+        return daysLeftToDownloadDoc;
```

Comment:
I'm just having a look at day.js documentation, and I think if we use some of their methods it will be a little simpler overall.

day.js has a few neat features that we can leverage so you don't have to write it all out, and is probably well tested (i.e. less for you to write ;)) https://day.js.org/docs/en/durations/diffing or https://day.js.org/docs/en/durations/subtract

Reply:
Good shout, actually if we start doing days and hours for some reason I think using their methods makes sense.